src/main/java/org/spongepowered/common/data/provider/item/stack/ItemStackDyeColorProvider.java

```diff
@@ -40,17 +41,19 @@ public ItemStackDyeColorProvider() {
         super(Keys.DYE_COLOR);
     }
 
-    /**
-     * Maybe try mixin injection in {@link Block.Properties#create(Material, net.minecraft.item.DyeColor)}
-     * to capture DyeColor.
-     */
     @Override
     protected Optional<DyeColor> getFrom(ItemStack dataHolder) {
-        return ((DyeColorItemBrige)dataHolder).bridge$getDyeColor();
+        Item item = dataHolder.getItem();
+        if (item instanceof BlockItem) {
+            Block block = ((BlockItem) item).getBlock();
+            net.minecraft.item.DyeColor color = ((DyeColorBlockBridge) block).bridge$getDyeColor();
+            return Optional.ofNullable((DyeColor) (Object) color);
```

Comment:
```suggestion
        final Item item = dataHolder.getItem();
        if (item instanceof BlockItem) {
            final Block block = ((BlockItem) item).getBlock();
            return ((DyeColorBlockBridge) block)..bridge$getDyeColor();
```
This way you have the optional already provided by the bridge implementation